integrations/terraform-cloud/client.py

```diff
@@ -0,0 +1,286 @@
+
+                    existing_configs_response = await self.client.get(notification_config_url)
+                    existing_configs_response.raise_for_status()
+                    existing_configs = existing_configs_response.json().get('data', [])
+
+                    for config in existing_configs:
+                        if config['attributes']['url'] == webhook_target_url:
+                            logger.info(f"Webhook already exists for workspace {workspace['id']}")
+                            break
+
+                    else:
+                        webhook_body = {
+                            "data": {
+                                "type": "notification-configurations",
+                                "attributes": {
+                                    "destination-type": "generic", 
+                                    "enabled": True,
+                                    "name": "port integration webhook",
+                                    "url": webhook_target_url,
+                                    "triggers": TERRAFORM_WEBHOOK_EVENTS
+                                }
+                            }
+                        }
+
+                        webhook_create_response = await self.client.post(notification_config_url, json=webhook_body)
+                        webhook_create_response.raise_for_status()
+                        logger.info(f"Webhook created for Terraform workspace {workspace['id']}")
+
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error for workspace {workspace.get('id', 'unknown')}: Status {e.response.status_code}, Response: {e.response.text}")
+        except httpx.HTTPError as e:
+            logger.error(f"Error fetching Terraform data: {e}")
+
+
+    async def get_current_state_version(self,workspace_id:str)-> [Dict[str, Any], None]:
+        
+        try:
+            
+            current_state_version_url = f"{self.api_url}/workspaces/{workspace_id}/current-state-version"
+            response = await self.client.get(current_state_version_url)
+            response.raise_for_status()
+            current_state_versions_data = response.json()
+            current_state_versions = current_state_versions_data.get('data', {})
+        
+            return current_state_versions
+        
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error for state versions: Status {e.response.status_code}, Response: {e.response.text}")
+        
+        except httpx.HTTPError as e:
+            logger.error(f"Error fetching Terraform data: {e}")
+    
+
+
+    async def get_state_version_for_workspace(self) -> AsyncGenerator[Dict[str, Any], None]:
+        try:
+
+            cache_key = CacheKeys.STATE_VERSION
+
+            if cache := event.attributes.get(cache_key):
+                logger.info("Fetching paginated workspaces from cache")
+                yield cache
+                return

```

Comment:
if you manage to get the output without triggering another request then please remove this cache

Reply:
get_paginated_workspaces is called by create_workspace_webhook, for workspace id's should the caching still be taken out ?

Reply:
yeah its a different event the cache wont be saved